aspnet/tutorials/web-api-help-pages-using-swagger.rst

```diff
@@ -0,0 +1,328 @@
+
+.. note:: For Linux or non-Windows operating systems, file names and paths can be case sensitive. So ``ToDoApi.xml`` would be found on Windows but not Centos for example.
+
+.. literalinclude:: web-api-help-pages-using-swagger/sample/src/TodoApi/Startup.cs
+    :language: c#
+    :start-after: snippet_Configure
+    :end-before: #endregion
+    :dedent: 8
+    :emphasize-lines: 29,32
+
+Adding the triple slash comments to the method enhances the Swagger UI by adding the description to the header of the section.
+
+.. literalinclude:: web-api-help-pages-using-swagger/sample/src/TodoApi/Controllers/TodoController.cs
+    :language: c#
+    :start-after: Delete_Method
+    :end-before: #endregion
+    :dedent: 8
+    :emphasize-lines: 2
+
+.. image:: web-api-help-pages-using-swagger/_static/triple-slash-comments.png
+
+Note that the UI is driven by the generated JSON file, and these comments are also in that file as well.
+
+.. code-block:: javascript
+    :emphasize-lines: 5
+
+      "delete": {
+        "tags": [
+          "Todo"
+        ],
+        "summary": "Deletes a specific TodoItem",
+        "operationId": "ApiTodoByIdDelete",
+        "consumes": [],
+        "produces": [],
+        "parameters": [
+          {
+            "name": "id",
+            "in": "path",
+            "description": "",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "No Content"
+          }
+        },
+        "deprecated": false
+      }
+
+Here is a more robust example, adding ``<remarks />`` where the content can be just text or adding the JSON or xml object for further documentation of the method.
+
+.. literalinclude:: web-api-help-pages-using-swagger/sample/src/TodoApi/Controllers/TodoController.cs
+    :language: c#
+    :start-after: Create_Method
+    :end-before: #endregion
+    :dedent: 8
+    :emphasize-lines: 4-14
+
+Notice the enhancement of the UI with these additional comments.
+
+.. image:: web-api-help-pages-using-swagger/_static/xml-comments-extended.png
+
+
+DataAnnotations
+'''''''''''''''
+Another supported feature is using the ``System.ComponentModel.DataAnnotations`` to decorate the API controller, methods and classes to help drive the Swagger components.
+
+For example, adding the ``[Required]`` annotation to the ``Name`` property of the TodoItem class will change the ModelSchema information in the UI. Other DataAnnotations such as ``[Produces("application/json"]``, RegularExpression validators and more will further detail the information delivered in the generated page.  What is important is that the more information that is documented in the code, the more "auto" generated information is pushed to the consumer.
```

Comment:
``[Produces("application/json"]``, regular expression validators and some other DataAnnotations  will be exposed in the generated page.